aws_kinesis_firehose_delivery_stream: Crash

### Terraform Version
1.7.1

### Affected Resource(s)
aws_kinesis_firehose_delivery_stream

### Terraform Configuration Files
```hcl
resource "aws_kinesis_firehose_delivery_stream" "firehose_delivery_stream" {
  name = "firehose-delivery-stream"
  destination = "s3"
  s3_configuration {
    role_arn = "${aws_iam_role.firehose_role.arn}"
    bucket_arn = "${aws_s3_bucket.firehose_delivery_bucket.arn}"
    buffer_size = 5
    buffer_interval = 300
    compression_format = "GZIP"
  }
}
```

### Panic Output
```
panic: runtime error: invalid memory address or nil pointer dereference
2018-01-19T11:04:27.699-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x27c986f]
2018-01-19T11:04:27.699-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4:
2018-01-19T11:04:27.699-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: goroutine 7792 [running]:
2018-01-19T11:04:27.699-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: github.com/terraform-providers/terraform-provider-aws/aws.flattenFirehoseS3Configuration(0xc42388c250, 0xc42388c260, 0x0, 0xc42388c270, 0xc42388c280, 0x0, 0xc42388c2a0, 0x151927d, 0x3, 0x348f99b)
2018-01-19T11:04:27.699-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_kinesis_firehose_delivery_stream.go:193 +0x16f
2018-01-19T11:04:27.699-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: github.com/terraform-providers/terraform-provider-aws/aws.flattenKinesisFirehoseDeliveryStream(0xc4218f6f50, 0xc420a872c0, 0xc420e04158, 0x0)
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_kinesis_firehose_delivery_stream.go:304 +0x151c
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsKinesisFirehoseDeliveryStreamRead(0xc4218f6f50, 0x3082ca0, 0xc4249c0a00, 0xc4218f6f50, 0x0)
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_kinesis_firehose_delivery_stream.go:1297 +0x323
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc424b70460, 0xc421d925a0, 0x3082ca0, 0xc4249c0a00, 0xc4217bc088, 0xc42557ea01, 0x80001082152)
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:354 +0x17e
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc4226d8310, 0xc421d92550, 0xc421d925a0, 0x5d4a8b0, 0x0, 0x2a5)
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:308 +0x9a
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc42057ef20, 0xc425ae1870, 0xc425ae1960, 0x0, 0x0)
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:549 +0x4e
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: reflect.Value.call(0xc42031ac60, 0xc4215a0998, 0x13, 0x3469695, 0x4, 0xc42557ef20, 0x3, 0x3, 0x0, 0x0, ...)
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: 	/usr/local/go/src/reflect/value.go:434 +0x905
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: reflect.Value.Call(0xc42031ac60, 0xc4215a0998, 0x13, 0xc4214cf720, 0x3, 0x3, 0xc421ba5f80, 0x33a6b80, 0xc4214cf750)
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: 	/usr/local/go/src/reflect/value.go:302 +0xa4
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: net/rpc.(*service).call(0xc420f5d440, 0xc42364b770, 0xc42017ea08, 0xc420633c00, 0xc423fb7ba0, 0x2c0dc40, 0xc425ae1870, 0x16, 0x2c0dc80, 0xc425ae1960, ...)
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: 	/usr/local/go/src/net/rpc/server.go:381 +0x142
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: created by net/rpc.(*Server).ServeCodec
2018-01-19T11:04:27.700-0700 [DEBUG] plugin.terraform-provider-aws_v1.7.1_x4: 	/usr/local/go/src/net/rpc/server.go:475 +0x36b
```

### Expected Behavior
plan file output

### Actual Behavior
Panic Crash

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. upgrade provider to 1.7.1
1. `terraform plan`


### References
- GH-3005

--- a/aws/resource_aws_kinesis_firehose_delivery_stream.go
+++ b/aws/resource_aws_kinesis_firehose_delivery_stream.go
@@ -190,7 +190,6 @@ func flattenFirehoseS3Configuration(s3 firehose.S3DestinationDescription) []inte
 	s3Configuration := map[string]interface{}{
 		"role_arn":           *s3.RoleARN,
 		"bucket_arn":         *s3.BucketARN,
-		"prefix":             *s3.Prefix,
 		"buffer_size":        *s3.BufferingHints.SizeInMBs,
 		"buffer_interval":    *s3.BufferingHints.IntervalInSeconds,
 		"compression_format": *s3.CompressionFormat,
@@ -200,6 +199,9 @@ func flattenFirehoseS3Configuration(s3 firehose.S3DestinationDescription) []inte
 	}
 	if s3.EncryptionConfiguration.KMSEncryptionConfig != nil {
 		s3Configuration["kms_key_arn"] = *s3.EncryptionConfiguration.KMSEncryptionConfig
+	}
+	if s3.Prefix != nil {
+		s3Configuration["prefix"] = *s3.Prefix
 	}
 	return []interface{}{s3Configuration}
 }

